runtime/moonbase/tests/integration_test.rs

```diff
@@ -40,6 +41,59 @@ fn fast_track_available() {
 	assert!(<moonbase_runtime::Runtime as pallet_democracy::Config>::InstantAllowed::get());
 }
 
+#[test]
+fn verify_pallet_prefixes() {
+	fn is_pallet_prefix<P: 'static>(name: &str) {
+		// Compares the unhashed pallet prefix in the `StorageInstance` implementation by every
+		// storage item in the pallet P. This pallet prefix is used in conjunction with the
+		// item name to get the unique storage key: hash(PalletPrefix) + hash(StorageName)
+		// https://github.com/paritytech/substrate/blob/master/frame/support/procedural/src/pallet/
+		// expand/storage.rs#L389-L401
+		assert_eq!(
+			<moonbase_runtime::Runtime as frame_system::Config>::PalletInfo::name::<P>(),
+			Some(name)
+		);
+	}
+	is_pallet_prefix::<moonbase_runtime::System>("System");
+	is_pallet_prefix::<moonbase_runtime::Utility>("Utility");
+	is_pallet_prefix::<moonbase_runtime::Timestamp>("Timestamp");
+	is_pallet_prefix::<moonbase_runtime::Balances>("Balances");
+	is_pallet_prefix::<moonbase_runtime::Sudo>("Sudo");
+	is_pallet_prefix::<moonbase_runtime::RandomnessCollectiveFlip>("RandomnessCollectiveFlip");
+	is_pallet_prefix::<moonbase_runtime::ParachainSystem>("ParachainSystem");
+	is_pallet_prefix::<moonbase_runtime::TransactionPayment>("TransactionPayment");
+	is_pallet_prefix::<moonbase_runtime::ParachainInfo>("ParachainInfo");
+	is_pallet_prefix::<moonbase_runtime::EthereumChainId>("EthereumChainId");
+	is_pallet_prefix::<moonbase_runtime::EVM>("EVM");
+	is_pallet_prefix::<moonbase_runtime::Ethereum>("Ethereum");
+	is_pallet_prefix::<moonbase_runtime::ParachainStaking>("ParachainStaking");
+	is_pallet_prefix::<moonbase_runtime::Scheduler>("Scheduler");
+	is_pallet_prefix::<moonbase_runtime::Democracy>("Democracy");
+	is_pallet_prefix::<moonbase_runtime::CouncilCollective>("CouncilCollective");
+	is_pallet_prefix::<moonbase_runtime::TechComitteeCollective>("TechComitteeCollective");
+	is_pallet_prefix::<moonbase_runtime::Treasury>("Treasury");
+	is_pallet_prefix::<moonbase_runtime::AuthorInherent>("AuthorInherent");
+	is_pallet_prefix::<moonbase_runtime::AuthorFilter>("AuthorFilter");
+	is_pallet_prefix::<moonbase_runtime::CrowdloanRewards>("CrowdloanRewards");
+	is_pallet_prefix::<moonbase_runtime::AuthorMapping>("AuthorMapping");
+	is_pallet_prefix::<moonbase_runtime::Proxy>("Proxy");
+	// TODO: some pallet instances do not impl StorageInfoTrait i.e. System, Utility
+	let prefix = |pallet_name, storage_name| {
+		let mut res = [0u8; 32];
+		res[0..16].copy_from_slice(&Twox128::hash(pallet_name));
+		res[16..32].copy_from_slice(&Twox128::hash(storage_name));
+		res
+	};
+	assert_eq!(
+		<moonbase_runtime::Sudo as StorageInfoTrait>::storage_info(),
+		vec![StorageInfo {
+			prefix: prefix(b"Sudo", b"Key"),
+			max_values: Some(1),
+			max_size: Some(20),
+		}]
+	);
```

Comment:
Unfortunately some pallet instances do not implement `StorageInfoTrait` so I cannot do this test for all pallets. I can remove it if we're satisfied with the above tests.

It still may be worth having some test like this for all storage items in all pallets to ensure the entire storage key does not change with any runtime changes. The tests further above only test the consistency of the pallet prefix part of the storage key.

Reply:
I like better the test where it queries from a storage item, as it is closer to the actual usage done by the runtime.
But both are good